workspace/exercise_metadata_collection.go

```diff
@@ -0,0 +1,18 @@
+package workspace
+
+// ExerciseMetadataCollection is a collection of exercise metadata to interactively choose from.
+type ExerciseMetadataCollection []*ExerciseMetadata
+
+// NewExerciseMetadataCollection loads up the exercise metadata for each of the provided paths.
+func NewExerciseMetadataCollection(paths []string) (ExerciseMetadataCollection, error) {
+	var metadataCollection []*ExerciseMetadata
```

Comment:
Seeing as you defined a custom type for a slice of ExerciseMetadata you should use that because while the following two variables are indeed a slice of ExerciseMetadata they are actually not the same type. Named types `ExerciseMetatadataCollection` are their own type, regardless of what their underlying type `[]*ExerciseMetadata` is. 

Here is an example of what I am talking about 
https://play.golang.org/p/_LuoH39ODUq

This concept is a little tough to explain in a review so take a minute to read the spec, and ask me any questions about it.  https://golang.org/ref/spec#Type_identity